fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonUtils.xtend

```diff
@@ -424,13 +429,18 @@ class PythonUtils {
 		    «protectedArea(cls, attribute.name + "_read", attribute.setAttrVal, false)»
 		    
     '''
-        
-    def readAttributeMethodHL(PogoDeviceClass cls, Attribute attribute) '''
-        def read_«attribute.name»(self):
+    def readAttributeMethodHL(PogoDeviceClass cls, Attribute attribute, boolean isDynamic) '''
+        def read_«attribute.name»(self«IF isDynamic», attr«ENDIF»):
                 «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
                 «openProtectedAreaHL(cls, attribute.name + "_read")»
+                «IF isDynamic==false»
                 """Return the «attribute.name» attribute."""
                 return self._«attribute.pythonAttributeVariableNameHL»
+                «ELSE»
+                """Return the «attribute.name»_read attribute."""
+                attr.set_value(self.attr_«attribute.name»_read)
+                return attr
```

Comment:
Since `self.attr_«attribute.name»_read` might not exist, suggest we comment out its use, and indicate it is an example.  We still need to return something - one option is the default value for the type of attribute, similar to line 446.
```suggestion
                """Example implementation:
                attr.set_value(self.attr_«attribute.name»_read)
                return attr
                """
                return «attribute.defaultValueDim»
```


This option might work too, although I'm not sure what value we will get from `attr` if we don't set it:
```suggestion
                """Example implementation:
                attr.set_value(self.attr_«attribute.name»_read)
                """
                return attr
```

Reply:
Thx for your answer.

Exactly, that I mean by:
> Currently what left to do is to get back to the previous solution to add the suggestion about the implementation of the dynamic attributes in the comment block (if this is accepted way by you).

And this is compliant with:
> Option 2: generate code that is commented out in both the initializer and the read/write implementations.

I will follow that way, and test generated device server,

Reply:
Hi,

After my tests I update this PR with working solution:

```python
    def read_dynamicImageAttribute(self, attr):
        # PROTECTED REGION ID(Test.dynamicImageAttribute_read) ENABLED START #
        """Return the dynamicImageAttribute_read attribute."""
        """Example implementation:
        Hint: self.attr_dynamicImageAttribute should be define on the top level
        (initialize_dynamic_attributes, init_device method or constructor)
        attr.set_value(self.attr_dynamicImageAttribute)
        return attr
        """
        attr.set_value([[0.0]])
        return attr
        # PROTECTED REGION END #    //  Test.dynamicImageAttribute_read

    def write_dynamicImageAttribute(self, w_attr):
        # PROTECTED REGION ID(Test.dynamicImageAttribute_write) ENABLED START #
        """Set the dynamicImageAttribute attribute."""
        """Example implementation:
        name = w_attr.get_name()
        Hint: self.attr_dynamicImageAttribute should be define on the top level
        (initialize_dynamic_attributes, init_device method or constructor)
        self.attr_dynamicImageAttribute = w_attr.get_write_value()
        """
        pass
        # PROTECTED REGION END #    //  Test.dynamicImageAttribute_write
```

attr.set_value([[0.0]]) this is a attr.set_value(«attribute.defaultValueDim»)

I'm open to your feedback and any information on what I can also improve.